src/System.Management.Automation/engine/Utils.cs

```diff
@@ -238,20 +238,35 @@ internal static string GetApplicationBaseFromRegistry(string shellId)
         }
 #endif
 
+        private static string s_GetApplicationBaseDefaultPowerShell = InitGetApplicationBaseDefaultPowerShell();
+
+        private static string InitGetApplicationBaseDefaultPowerShell()
+        {
+            return GetApplicationBase(Utils.DefaultPowerShellShellID);
+        }
+
```

Comment:
How about having an internal static property named `DefaultPowerShellAppBase` like this:
```CSharp
internal static string DefaultPowerShellAppBase { get; } = GetApplicationBase(DefaultPowerShellShellID);
```
And then use this get-only property in places where `GetApplicationBase(DefaultPowerShellShellID)` is used.